### PR TITLE
Sidekiq course delete#1013

### PR DESCRIPTION
**Pull request by patmbolger**
Convert from sync to async operation via sidekiq: Deleting course from CoursesController#destroy

### CODE REVIEW

**Comment by ragesoss** [comment]
Yikes. I didn't think this one through. We can't do this async as-is, because the course will already be destroyed by the time sidekiq gets to it.

We should either keep this one syncronous, or destroy the course inside the worker. I *think* the latter would be fine.

**Comment by ragesoss** [comment]
Yeah, destroying the course within the worker should be fine.

**Comment by patmbolger** [comment]
Yes, I swapped the order of actions so that the local course is destroyed *after* the async job is scheduled.  Unless we think there is a chance that the course `destroy` will fail, I did not see much difference between doing it like this or performing the `destroy` within the worker.

Do you still want to do it in the worker?

**Comment by ragesoss** [comment]
It would fail when the worker tried to perform, because the course would already be gone from the database. You could schedule it, but looking up the course by ID would cause an error. (Doesn't surface in the test, because sidekiq runs in-thread syncronously in the test suite.)

**Comment by patmbolger** [comment]
Ah yes, of course.  Good point.

File: app/controllers/courses_controller.rb [hunk 1]
```diff
@@ -46,11 +46,7 @@ def update
 
   def destroy
     validate
-    @course.destroy
-    WikiCourseEdits.new(action: :update_course,
-                        course: @course,
-                        current_user: current_user,
-                        delete: true)
+    DeleteCourseWorker.schedule_edits(course: @course, current_user: current_user)
```
Comment (ragesoss) [inline app/controllers/courses_controller.rb]: Since this does more than just delete it, let's name the method differently. `schedule_deletion` perhaps?

Comment (ragesoss) [inline app/controllers/courses_controller.rb]: Change name here too.

**Pull request merged**